### PR TITLE
Updates: ruby 2.7.6, some gems, and CircleCI config

### PR DESCRIPTION
**Pull request by tiegz**
* ruby 2.6.5 -> 2.7.6
* update some gems to fix specs: charlock_holmes, nokogiri, groupdate, pry and rugged
* removes "build" section from CI config bc it wasn't doing anything
* update CI config to use newest ruby orb
* remove the caching logic in CI config to use the builtin ruby orb commands

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -20,48 +20,18 @@ executors:
...
-      - run: sudo apt-get -y -qq update
-      - run: sudo apt-get -y --no-install-recommends install cmake libpq-dev
-      - ruby/bundle-install
-      - save_cache:
-          key: v1-gem-cache-{{ arch }}-{{ .Branch }}-{{ checksum "Gemfile.lock" }}
-          paths:
-            - vendor/bundle
-      - ruby/test
+      - run: apt-get -y -qq update
+      - run: apt-get -y --no-install-recommends install cmake libpq-dev git-core build-essential libicu-dev pkg-config nodejs
```
Comment (tiegz) [inline .circleci/config.yml:29]: Either the ruby container or the circle ruby orb changed such that we have a few more dep requirements in here now.

File: Gemfile [hunk 2]
```diff
@@ -74,7 +74,6 @@ gem "rails", "5.2.6.2"
 gem "rails-timeago", git: "https://github.com/librariesio/rails-timeago"
 gem "rb-readline"
 gem "rdoc"
-gem "RedCloth"
```
Comment (tiegz) [inline Gemfile:77]: Couldn't find a reference to RedCloth being used, even tho we do consume Textile in readmes.

File: .circleci/config.yml [hunk 3]
```diff
@@ -20,48 +20,18 @@ executors:
...
-            - v1-gem-cache-{{ arch }}-{{ .Branch }}-{{ checksum "Gemfile.lock" }}
-            - v1-gem-cache-{{ arch }}-{{ .Branch }}-
-            - v1-gem-cache-{{ arch }}-
-      - run: sudo apt-get -y -qq update
-      - run: sudo apt-get -y --no-install-recommends install cmake libpq-dev
-      - ruby/bundle-install
-      - save_cache:
-          key: v1-gem-cache-{{ arch }}-{{ .Branch }}-{{ checksum "Gemfile.lock" }}
-          paths:
-            - vendor/bundle
```
Comment (tiegz) [inline .circleci/config.yml:39]: Couldn't figure out a reason why we'd need the `build` section, when it's almost identical to the test step.

File: .circleci/config.yml [hunk 4]
```diff
@@ -20,48 +20,18 @@ executors:
...
```
Comment (tiegz) [inline .circleci/config.yml:57]: The new orb version does this caching for us now.

**Review by djpowers — APPROVED** [review]

**Pull request merged**